exporters/prometheus/exporter.go

```diff
@@ -140,48 +146,60 @@ func (c *collector) Collect(ch chan<- prometheus.Metric) {
 		var keys, values [2]string
 
 		if !c.disableScopeInfo {
-			scopeInfo, ok := c.scopeInfos[scopeMetrics.Scope]
-			if !ok {
-				scopeInfo, err = createScopeInfoMetric(scopeMetrics.Scope)
-				if err != nil {
-					otel.Handle(err)
+			func() {
+				c.mu.Lock()
+				defer c.mu.Unlock()
+
+				scopeInfo, ok := c.scopeInfos[scopeMetrics.Scope]
+				if !ok {
+					scopeInfo, err = createScopeInfoMetric(scopeMetrics.Scope)
+					if err != nil {
+						otel.Handle(err)
+					}
+					c.scopeInfos[scopeMetrics.Scope] = scopeInfo
 				}
-				c.scopeInfos[scopeMetrics.Scope] = scopeInfo
-			}
-			ch <- scopeInfo
+				ch <- scopeInfo
```

Comment:
I was going to fix this because it seems like a bug to me.

Reply:
It would be better to address in a separate PR. I created an issue https://github.com/open-telemetry/opentelemetry-go/issues/3917. Thanks for your help 👍

Reply:
> It would be better to address in a separate PR. I created an issue #3917. Thanks for your help +1

I have to fix the very same section.

Reply:
@pellared I fixed this too. Kind of understand that it would be better to address this in a separate PR but I'm working on these exact lines of code so it would be weird to leave this as is.